Classes/Middleware/RedirectMiddleware.php

```diff
@@ -38,13 +39,21 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         $site = $request->getAttribute('site');
+        if ($site instanceof NullSite) {
+            return $handler->handle($request);
+        }
         assert($site instanceof Site);
         foreach ($site->getLanguages() as $language) {
             if ($language->getBase()->getHost() !== $request->getUri()->getHost()) {
                 continue;
             }
 
-            if ($language->getBase()->getPath() !== $request->getUri()->getPath()) {
+            if ($request->getUri()->getPath() === '/' && $language->getBase()->getPath() === '') {
+                return $handler->handle($request);
+            }
```

Comment:
```suggestion
            if ($language->getBase()->getPath() === '') {
                // if there is a language without a path, we should not redirect to any other langauge.
                return $handler->handle($request);
            }
```

the uri is always `'/'` see line 36.